SPECS/systemd/systemd.spec

```diff
@@ -47,6 +48,7 @@ BuildRequires:  tpm2-tss-devel
 BuildRequires:  util-linux-devel
 BuildRequires:  xz-devel
 Requires:       %{name}-rpm-macros = %{version}-%{release}
+Requires:       audit
```

Comment:
We intentionally don't have audit installed on a few configurations. Can we remove this Requires and still have systemd operate both with and without audit installed?

Reply:
That's a good point. If we build with `audit-devel` and `-Daudit=true` we will most probably end up with binaries having a dependency on dynamic libraries from `audit`. Depending on the binaries now loading `audit` libs during run-time we may or may not be able to break the dependency. If the `systemd` main executable ends up with the new dependency, then this might be unbreakable. If only some non-critical tools/executables have the new dependency, then we would have the option of moving them to a separate subpackage, which would keep the default `systemd` clean. I wouldn't be surprised, however, if it's going to turn out, that the dependency is unbreakable this way.

An alternative would be to now build `systemd` twice with and without the `-Daudit=true` switch and put the resulting binaries in separate subpackages.

Reply:
Looks like systemd happily ignores failures to connect the audit socket.

There is a dependency on `audit-libs`. But that dependency has already been declared.

Reply:
And is there still no implicit (added by RPM) dependency between `systemd` and `audit` after this change? This should be visible by comparing the output of `rpm -q --requires <systemd_rpm>` before and after the change.

Reply:
Running that command showed no dependency on the `audit` package. (Also, during testing I was able to remove `audit` without `dnf` wanting to remove `systemd`.)

There is a dependency on `libaudit.so`. But that is provided by `audit-libs` not `audit.`